[CI] Failure in org.elasticsearch.cluster.routing.AllocationIdIT.testFailedRecoveryOnAllocateStalePrimaryRequiresAnotherAllocateStalePrimary

This just failed on 7.x here: https://gradle-enterprise.elastic.co/s/rhyopb5hm3gu2/tests/:server:internalClusterTest/org.elasticsearch.cluster.routing.AllocationIdIT/testFailedRecoveryOnAllocateStalePrimaryRequiresAnotherAllocateStalePrimary

```
  2> REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.cluster.routing.AllocationIdIT.testFailedRecoveryOnAllocateStalePrimaryRequiresAnotherAllocateStalePrimary" -Dtests.seed=491F1EE43B42DEC9 -Dtests.security.manager=true -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dtests.locale=he-IL -Dtests.timezone=Asia/Damascus -Druntime.java=8
  2> java.lang.AssertionError: timed out waiting for yellow state
        at __randomizedtesting.SeedInfo.seed([491F1EE43B42DEC9:C099E915F71965EB]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.test.ESIntegTestCase.ensureColor(ESIntegTestCase.java:953)
        at org.elasticsearch.test.ESIntegTestCase.ensureYellow(ESIntegTestCase.java:913)

```

does not reproduce locally.

--- a/server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/AllocationIdIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/AllocationIdIT.java
@@ -136,17 +136,13 @@ public class AllocationIdIT extends ESIntegTestCase {
             assertThat(shardRouting.unassignedInfo().getReason(), equalTo(UnassignedInfo.Reason.ALLOCATION_FAILED));
         });
 
+        internalCluster().stopNode(node1);
         try(Store store = new Store(shardId, indexSettings, new SimpleFSDirectory(indexPath), new DummyShardLock(shardId))) {
             store.removeCorruptionMarker();
         }
+        node1 = internalCluster().startNode(node1DataPathSettings);
 
         // index is red: no any shard is allocated (allocation id is a fake id that does not match to anything)
-        checkHealthStatus(indexName, ClusterHealthStatus.RED);
-        checkNoValidShardCopy(indexName, shardId);
-
-        internalCluster().restartNode(node1, InternalTestCluster.EMPTY_CALLBACK);
-
-        // index is still red due to mismatch of allocation id
         checkHealthStatus(indexName, ClusterHealthStatus.RED);
         checkNoValidShardCopy(indexName, shardId);
 

